### DISCUSSION TITLE
Custom password field properties

### DISCUSSION THREAD

**Discussion by jmikrut** [payloadcms/payload#1713] [Roadmap]
We should allow developers to declare a custom Password field on auth-enabled collections, wherein properties such as `validate` and similar are merged into our base Password field.

This would be useful to create custom password validations, etc.

**Comment by iakshay** [comment]
On a related note: It seems like there is no way to opt out of password strategy and if one implements a custom strategy one has to use a different collection for authentication or populate a random password. As in the google-one-tap strategy here - 

https://github.com/ScottEAdams/payload-plugin-google-one-tap/blob/main/src/payload/strategy.ts#L49

If a user is created with a strategy X, ideally we would also want to prevent them from logging in with password. I guess this might be possible with `beforeLogin` hook.

Would be nice to clean things up here, so password is the default strategy but still possible to opt-out of it.

**Reply by DanRibbens** [reply]
Did you see `disableLocalStrategy` in https://payloadcms.com/docs/authentication/config#options?

That lets you turn off the password based auth managed by Payload when building your own authentication for users and still be recognized by the admin UI.

**Reply by iakshay** [reply]
That works, is it possible to make the password field optional? So one does not need to generate random password for custom strategy while still using the builtin local strategy.

Maybe we make `enabledStrategies` a mandatory field on `auth` collection? And only if strategy includes `local` - password is required. 

Everything is possible today, I'm thinking of possible refinements to make things cleaner.

**Reply by DanRibbens** [reply]
You can override the validate function of the password field. To do that you have to add the field on your collection instead of letting Payload make it for you.

‘‘‘js
fields: [{
  name: 'password',
  type: 'password',
  validate: () = true,
}]
‘‘‘

Does that work for you?

**Reply by arsekil** [reply]
This way, can I write a custom validation hook or function for `validate`?

**Reply by MarvinVrdoljak** [reply]
> This way, can I write a custom validation hook or function for `validate`?

I think you can't. This is only an idea they didn't implemented yet.

**Reply by DanRibbens** [reply]
Custom validation is fully implemented and documented here: https://payloadcms.com/docs/fields/overview#validation

**Reply by MarvinVrdoljak** [reply]
Yes, but can't validate a user's password like this or am I wrong? You can't even set a new password field. Type password is not a valid field type.

**Reply by josa-languagewire** [reply]
I'm interested in a field type "password" too but I can find nothing in the Payload documentation or discord.
Any hints?

**Reply by MarvinVrdoljak** [reply]
@josa-languagewire Write your own custom field: https://payloadcms.com/blog/building-a-custom-field

**Reply by josa-languagewire** [reply]
I know... this is always an option, but I just find it weird there is not a solution already for such a common use case.
I field type "secret/password" should be handling both encryption and not showing the value but allowing to update it for a new one in some cases...

**Reply by mckenzie-boyd** [reply]
Very surprised this is not yet implemented. Would be such a useful feature.
